io/include/traccc/io/csv/measurement.hpp

```diff
@@ -21,7 +21,7 @@ struct measurement {
 
     uint64_t measurement_id = 0;
     uint64_t geometry_id = 0;
-    std::string local_key = "";
+    uint8_t local_key = 0;
```

Comment:
Are you sure that a `uint16_t` would not serve us better here? The `uint8_t` type is clearly handled as a `char` by the I/O. Leading to some awkward characters in the CSV files.

I think the larger memory footprint of `uint16_t` is overruled by its better ease of use of. :wink:

Reply:
It's awkward indeed, but that's what is defined in ACTS Core repository: https://github.com/acts-project/acts/blob/ea7aaa5086ebd3618360949df04f885488d3d9ce/Examples/Io/Csv/src/CsvOutputData.hpp#L123

I think we need to keep the same type to read ACTS simulation data properly.

6 bits in the middle are used for each index of bound track parameter (x,y,phi,theta,q/p,t); 1 means the measurement contains the information of the parameter with corresponding index.

If I am not wrong, the first and last digit are dummy values

Reply:
Okay, that's an acceptable reason for the type then. Objection withdrawn.

Reply:
> If `uint16_t` is be compatible with `uint8_t` in I/O-wise, then I think it is good to change

No, I think it would hurt us. I just didn't think of the necessary compatibility with Acts...